app/controllers/TransportController.scala

```diff
@@ -41,14 +42,18 @@ class TransportController @Inject()(
     extends FrontendController(mcc) with I18nSupport {
 
   def displayPage(): Action[AnyContent] = (authenticate andThen getJourney(JourneyType.DEPART)) { implicit request =>
-    Ok(transportPage(request.answersAs[DepartureAnswers].transport.fold(form)(form.fill(_))))
+    val answers = request.answersAs[DepartureAnswers]
+    val consignmentReference = answers.consignmentReferences.map(_.referenceValue)
+    Ok(transportPage(answers.transport.fold(form)(form.fill), consignmentReference))
   }
 
   def saveTransport(): Action[AnyContent] = (authenticate andThen getJourney(JourneyType.DEPART)).async { implicit request =>
+    val answers = request.answersAs[DepartureAnswers]
+    val consignmentReference = answers.consignmentReferences.map(_.referenceValue)
     form
       .bindFromRequest()
       .fold(
-        (formWithErrors: Form[Transport]) => Future.successful(BadRequest(transportPage(formWithErrors))),
+        (formWithErrors: Form[Transport]) => Future.successful(BadRequest(transportPage(formWithErrors, consignmentReference))),
         validForm => {
```

Comment:
You could have extracted consignmentReference into a method (as you started to do) as here, in particular, it only needs to be called if there is an error on the submitted form.
But this is ok as well.